example/main.py

```diff
@@ -1,163 +1,142 @@
-        x = self.fc3(x)
-        return x
-
-
-def get_dataset(dataset_name, transform, batch_size=64):
+def get_dataset(args, transform):
     """
     :param dataset_name:
     :param transform:
     :param batch_size:
     :return: iterators for the dataset
     """
-    if dataset_name == 'MNIST':
+    if args.dataset == 'MNIST':
         trainset = torchvision.datasets.MNIST(root='./data', train=True, download=True, transform=transform)
         testset = torchvision.datasets.MNIST(root='./data', train=False, download=True, transform=transform)
     else:
         trainset = torchvision.datasets.CIFAR10(root='./data', train=True, download=True, transform=transform)
         testset = torchvision.datasets.CIFAR10(root='./data', train=False, download=True, transform=transform)
 
-    trainloader = torch.utils.data.DataLoader(trainset, batch_size=batch_size, shuffle=True, num_workers=1)
-    testloader = torch.utils.data.DataLoader(testset, batch_size=batch_size, shuffle=False, num_workers=1)
+    trainloader = torch.utils.data.DataLoader(trainset, batch_size=args.batch_size, shuffle=True, num_workers=1)
+    testloader = torch.utils.data.DataLoader(testset, batch_size=args.test_batch_size, shuffle=False, num_workers=1)
     return trainloader, testloader
 
-
-def main(*args, **kwargs):
-    parser = argparse.ArgumentParser(description='Distbelief training example')
-    parser.add_argument('--batch-size', type=int, default=64, metavar='N',
-                        help='input batch size for training (default: 64)')
-    parser.add_argument('--test-batch-size', type=int, default=1000, metavar='N',
-                        help='input batch size for testing (default: 1000)')
-    parser.add_argument('--epochs', type=int, default=10, metavar='N', help='number of epochs to train (default: 10)')
-    parser.add_argument('--lr', type=float, default=0.01, metavar='LR', help='learning rate (default: 0.01)')
-    parser.add_argument('--momentum', type=float, default=0.5, metavar='M', help='SGD momentum (default: 0.5)')
-    parser.add_argument('--no-cuda', action='store_true', default=False, help='disables CUDA training')
-    parser.add_argument('--seed', type=int, default=1, metavar='S', help='random seed (default: 1)')
-    parser.add_argument('--log-interval', type=int, default=10, metavar='N',
-                        help='how many batches to wait before logging training status')
-    parser.add_argument('--dataset', type=str, default='MNIST', help='Dataset to train with (CIFAR or MNIST)')
-
-    args = parser.parse_args()
+def main(args):
 
     transform = transforms.Compose([
                 transforms.ToTensor(),
                 transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
             ])
 
-    trainloader, testloader = get_dataset(args.dataset, transform)
-
-    net = Net(num_channels=1 if args.dataset=='MNIST' else 3)
+    trainloader, testloader = get_dataset(args, transform)
+    net = LeNet()
```

Comment:
I don't think the current implementation of `LeNet` in the models file will work with the MNIST dataset, since MNIST has 1 channel while the first conv step in LeNet appears to expect 3 channels. The previous code looked at the number of channels in the input dataset to get the dims right for both `MNIST` and `CIFAR`.

Reply:
ah kk will change back

Reply:
should we just take out MNIST support? i feel like it's not as useful as cifar

Reply:
I think it's worth it to support `MNIST` just as a quicker way of getting a benchmark. If you want, I can go ahead and add back the support once you merge.